src/coreclr/src/vm/ceeload.cpp

```diff
@@ -9036,9 +9036,9 @@ void Module::PlaceType(DataImage *image, TypeHandle th, DWORD profilingFlags)
 
         if (profilingFlags & (1 << ReadFieldMarshalers))
         {
-            if (pClass->HasLayout() && pClass->GetLayoutInfo()->GetNumCTMFields() > 0)
+            if (pClass->HasLayout())
             {
-                image->PlaceStructureForAddress((void *)pClass->GetLayoutInfo()->GetNativeFieldDescriptors(), CORCOMPILE_SECTION_HOT);
+                image->PlaceStructureForAddress((void *)pClass->GetNativeLayoutInfo(), CORCOMPILE_SECTION_HOT);
```

Comment:
I think you can delete all code that has to do with saving m_nativeLayoutInfo into fragile NGen images. It does not make sense to be saving now that we can recreate it on demand at runtime.